### ISSUE TITLE
--arch x86_omf tries to use LLD to link OMF object files on Windows

### ISSUE THREAD

**Issue by AndrejMitrovic** [dlang/dub#2393]
### System information
DUB version 1.29.0, built on May 13 2022
Win10 x64
DMD v2.100.0

### Bug Description

Expecting `dub -v build -a x86_omf` to build OMF object files and then use Optlink to link them.

Instead, dub uses LLD to try to link the object files:

```
C:\D\dmd2\windows\bin\dmd.exe -m32omf -c -of.dub\build\HelloMsg-debug-windows-x86.x86_omf-dmd_v2.100.0-dirty-5EE6860E0579DE7834800F6D4968E9FA\HelloMsg.obj -debug -g -w -version=Unicode -version=WindowsXP -version=Have_dwinprogramming Samples\Chap01\HelloMsg\HelloMsg.d -vcolumns

Linking...

C:\D\dmd2\windows\bin\dmd.exe -of.dub\build\HelloMsg-debug-windows-x86.x86_omf-dmd_v2.100.0-dirty-5EE6860E0579DE7834800F6D4968E9FA\HelloMsg.exe .dub\build\HelloMsg-debug-windows-x86.x86_omf-dmd_v2.100.0-dirty-5EE6860E0579DE7834800F6D4968E9FA\HelloMsg.obj -g

lld-link: error: .dub\build\HelloMsg-debug-windows-x86.x86_omf-dmd_v2.100.0-dirty-5EE6860E0579DE7834800F6D4968E9FA\HelloMsg.obj: unknown file type

Error: linker exited with status 1

FAIL .dub\build\HelloMsg-debug-windows-x86.x86_omf-dmd_v2.100.0-dirty-5EE6860E0579DE7834800F6D4968E9FA\ HelloMsg executable
```

### Workaround

Use the `--rdmd` flag to build:

```
dub -v build --rdmd -a x86_omf
```

Note that `x86_omf` is explicitly listed in the supported options for `--arch` on https://dub.pm/package-format-json.html#build-settings.

Since `-m32omf` is deprecated on DMD maybe it's best to just remove support for OMF in dub if it doesn't work, and update the documentation?

**Comment by rikkimax** [comment]
Possibly fixed by #2389.

**Comment by AndrejMitrovic** [comment]
It is! How lucky it got fixed just hours before it was reported. :D
